internal/zinc-classpath/src/main/scala/sbt/internal/inc/classpath/ClassLoaders.scala

```diff
@@ -58,6 +58,14 @@ final class SelfFirstLoader(classpath: Seq[URL], parent: ClassLoader)
 /** Doesn't load any classes itself, but instead verifies that all classes loaded through `parent` either come from `root` or `classpath`.*/
 final class ClasspathFilter(parent: ClassLoader, root: ClassLoader, classpath: Set[File])
     extends ClassLoader(parent) {
+
+  def close(): Unit = {
```

Comment:
Why this new addition?

Reply:
It is called at `sbt/internal/inc/IncHandler.scala:179`. This is similar case to the other classloader issue. We make a classloader to run main method, but then we never close it. Every resource should be eventually released, regardless if it actually causes issues, but this actually did cause issues.
AFAIR after running a main class from jar, URLClassLoader was holding the jar and it could not be deleted on Windows. As multiple scripted tests are run in the same directory, the delete has to work or tests will start to behave incorrectly.